llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -54,9 +73,15 @@ TGLexer::TGLexer(SourceMgr &SM, ArrayRef<std::string> Macros) : SrcMgr(SM) {
   PrepIncludeStack.push_back(
       std::make_unique<std::vector<PreprocessorControlDesc>>());
 
-  // Put all macros defined in the command line into the DefinedMacros set.
-  for (const std::string &MacroName : Macros)
+  // Add all macros defined on the command line to the DefinedMacros set.
+  // Check invalid macro names and print fatal error if we find one.
+  for (const std::string &MacroName : Macros) {
```

Comment:
```suggestion
  for (StringRef MacroName : Macros) {
```

Reply:
N/A anymore.

Reply:
This isn't N/A, it should still be StringRef.

Reply:
Right, I have changed it. And it makes the code simpler as I can change the return pointer from `lexMacroName` directly against MacroName.end().